src/ModelContextProtocol/Protocol/Types/ToolAnnotations.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Text.Json.Serialization;
+
+namespace ModelContextProtocol.Protocol.Types;
+
+/// <summary>
+/// Additional properties describing a Tool to clients.
+/// NOTE: all properties in ToolAnnotations are **hints**.
+/// They are not guaranteed to provide a faithful description of tool behavior (including descriptive properties like `title`).
+/// Clients should never make tool use decisions based on ToolAnnotations received from untrusted servers.
+/// <see href="https://github.com/modelcontextprotocol/specification/blob/main/schema/">See the schema for details</see>
+/// There are multiple subtypes of content, depending on the "type" field, these are represented as separate classes.
+/// </summary>
+public class ToolAnnotations
+{
+    /// <summary>
+    /// A human-readable title for the tool.
+    /// </summary>
+    [JsonPropertyName("title")]
+    public string? Title { get; set; }
```

Comment:
The models are somewhat inconsistent on whether they use init or set. I take it we're moving to making everything mutable eventually?

Reply:
We can go through them all to move them consistently to one approach or the other. My preference is for mutable, just because it makes things easier to work with. But if we want to go the other way, that's fine. As you say, we're already inconsistent.